pkg/graveler/staging/manager.go

```diff
@@ -4,112 +4,106 @@ import (
-	if errors.Is(err, db.ErrNotFound) {
-		return nil, graveler.ErrNotFound
+func protoFromValue(key []byte, m *graveler.Value) *graveler.StagedObject {
+	return &graveler.StagedObject{
+		Key:      key,
+		Identity: m.Identity,
+		Data:     m.Data,
+	}
+}
+
+func NewManager(store kv.StoreMessage) *Manager {
+	return &Manager{
+		store: store,
+		log:   logging.Default().WithField("service_name", "staging_manager"),
 	}
+}
+
+func (m *Manager) Get(ctx context.Context, st graveler.StagingToken, key graveler.Key) (*graveler.Value, error) {
+	data := &graveler.StagedObject{}
+	_, err := m.store.GetMsg(ctx, string(st), key, data)
 	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrNotFound
+		}
 		return nil, err
 	}
-	value := res.(*graveler.Value)
-	if value.Identity == nil {
+	// Tombstone handling
+	if data.Identity == nil {
 		return nil, nil
 	}
-	return value, nil
+	return valueFromProto(data), nil
 }
 
-func (p *Manager) Set(ctx context.Context, st graveler.StagingToken, key graveler.Key, value *graveler.Value, overwrite bool) error {
+func (m *Manager) Set(ctx context.Context, st graveler.StagingToken, key graveler.Key, value *graveler.Value, overwrite bool) error {
+	// Tombstone handling
 	if value == nil {
 		value = new(graveler.Value)
 	} else if value.Identity == nil {
 		return graveler.ErrInvalidValue
 	}
-	_, err := p.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		if !overwrite {
-			res, err := tx.Exec(
-				`INSERT INTO graveler_staging_kv (staging_token, key, identity, data)
-						VALUES ($1, $2, $3, $4)
-						ON CONFLICT (staging_token, key) DO NOTHING`,
-				st, key, value.Identity, value.Data)
-			if err != nil {
-				return nil, err
-			}
-			if res.RowsAffected() == 0 {
-				return nil, graveler.ErrPreconditionFailed
-			}
-			return res, err
+
+	var err error
+	pb := protoFromValue(key, value)
+	if overwrite {
+		err = m.store.SetMsg(ctx, string(st), key, pb)
+	} else {
+		err = m.store.SetMsgIf(ctx, string(st), key, pb, nil)
+		if errors.Is(err, kv.ErrPredicateFailed) {
+			return graveler.ErrPreconditionFailed
 		}
-		return tx.Exec(`INSERT INTO graveler_staging_kv (staging_token, key, identity, data)
-								VALUES ($1, $2, $3, $4)
-								ON CONFLICT (staging_token, key) DO UPDATE
-									SET (staging_token, key, identity, data) =
-											(excluded.staging_token, excluded.key, excluded.identity, excluded.data)`,
-			st, key, value.Identity, value.Data)
-	}, p.txOpts()...)
+	}
 	return err
 }
 
-func (p *Manager) DropKey(ctx context.Context, st graveler.StagingToken, key graveler.Key) error {
-	_, err := p.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		return tx.Exec("DELETE FROM graveler_staging_kv WHERE staging_token=$1 AND key=$2", st, key)
-	}, p.txOpts()...)
+func (m *Manager) DropKey(ctx context.Context, st graveler.StagingToken, key graveler.Key) error {
+	// Changed behavior from DB - we will not get an error on delete non-existent
```

Comment:
Correctness maintained from the lakeFS side because the metadata state stays the same. But consumers of this layer might expect the previous error so we need to check that this change of behaviour is accepted by them.